### ISSUE TITLE
git credential.helper custom helper shell commands are broken

### ISSUE THREAD

**Issue by JonVDB** [desktop/desktop#19015]
### The problem

In my global gitconfig, I have configured `credential.helper` as a custom helper shell snippet (using the `!` prefix) to connect with my AWS CodeCommit repository. The specific command is `!aws codecommit credential-helper $@` .
I've paired this with the configuration `credential.UseHttpPath` with value `true`.

This functions perfectly fine using Git in the CLI.

Using GitHub Desktop, authentication fails and the username/password window for authentication keeps popping up. It seems like GitHub Desktop just ignores the `credential.helper` configuration. I conclude this because configuring random commands as `credential.helper` also doesn't work. For example, on Windows I can configure the helper with value `!explorer.exe .` which, when running Git from the CLI, will make the Windows Explorer pop up, while doing so through GitHub Desktop will not.

### Release version

3.4.2 (x64)

### Operating system

Win11Pro22H2

### Steps to reproduce the behavior

1. Set up an AWS CodeCommit repo
2. Set up an AWS user with the necessary IAM-permissions to access the repo - do not use CodeCommit credentials
3. Set up your AWS CLI with the necessary default-profile credentials & configs to be authenticated with AWS
4. Configure your global gitconfig with the following
- `credential.helper` with value `!aws codecommit credential-helper $@`
- `credential.UseHttpPath` with value `true`
5. In GitHub Desktop, try to clone/fetch/pull/push to the repo. The "Authentication failed" username/password screen will pop up.

### Log files

```
2024-07-25T09:16:00.277Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 2.360s)
2024-07-25T09:20:48.849Z - error: [ui] `git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin` exited with an unexpected code: 128.
stderr:
fatal: Authentication failed for 'https://git-codecommit.eu-west-1.amazonaws.com/v1/repos/[redacted]/'

(The error was parsed as 3: Authentication failed. Some common reasons include:

- You are not logged in to your account: see File > Options.
- You may need to log out and log back in to refresh your token.
- You do not have permission to access this repository.
- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
- If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account.
- If you use SSH authentication, ensure the host key verification passes for your repository hosting service.
- If you used username / password authentication, you might need to use a Personal Access Token instead of your account password. Check the documentation of your repository hosting service.)
```

### Additional context

Due to security restraints it's necessary to use this specific method of authenticating with the AWS CodeCommit repo.

In general it would be nice for GitHub Desktop to at least use `credential.helper` and execute the command if one has been provided. In situations like these, no other pop-ups or redirects are required, only the execution of that command.

**Comment by tidy-dev** [comment]
@JonVDB Thank you for reaching out. GitHub Desktop now uses Git Credential Manager. In order to do so, we have to clear out the credential.helper git config during authentication. 

Would it be possible to use Git Credential Manager to authenticate with AWS CodeCommit? 

If so, could you try enabling the Git Credential Manager for repositories not hosted on github.com? Options > Advanced "Use Git Credential Manager".

**Comment by JonVDB** [comment]
@tidy-dev Thank you for the quick reply! I had already looked into Git Credential Manager. Unfortunately, to my understanding, it does not allow to pass along the command that I currently put into `credential.helper` and it requires its own value for that property (`manager`).

Fortunately I've found a work-around using a Python tool that hooks straight into Git, and that seems to work well in combination with GitHub Desktop!

I'll close this Issue for now as it is obsolete with my current fix. Apologies for the trouble!
For anyone struggling with the same issue; do look into the **git-remote-codecommit** AWS tool.
